### ISSUE TITLE
[Troubleshoot]: Cannot reproduce evaluation results

### ISSUE THREAD

**Issue by Weott** [RLinf/RLinf#75]
### Problem description

Hello,

I'm encountering an issue where I cannot reproduce the expected evaluation results for several models.

I tested the following models using the `libero_10_grpo_openvlaoft_eval.yaml` configuration with the command:
`bash examples/embodiment/eval_embodiment.sh libero_10_grpo_openvlaoft_eval`

The models tested are:

1. Openvla-oft-SFT-libero10-traj1
2. Openvla-oft-SFT-libero10-trajall
3. RLinf-OpenVLAOFT-GRPO-LIBERO-long

The evaluation results I obtained are:

[libero_10_grpo_openvlaoft_eval.yaml](https://github.com/user-attachments/files/22206922/libero_10_grpo_openvlaoft_eval.yaml)

1. For Openvla-oft-SFT-libero10-traj1:
`eval_metrics={'eval/env_info/success_once': array(0., dtype=float32), 'eval/env_info/return': array(0., dtype=float32), 'eval/env_info/episode_len': array(512., dtype=float32), 'eval/env_info/reward': array(0., dtype=float32), 'eval/env_info/success_at_end': array(0., dtype=float32)}`

2. For Openvla-oft-SFT-libero10-trajall:
`eval_metrics={'eval/env_info/success_once': array(0.2578125, dtype=float32), 'eval/env_info/return': array(1.09375, dtype=float32), 'eval/env_info/episode_len': array(512., dtype=float32), 'eval/env_info/reward': array(0.00213623, dtype=float32), 'eval/env_info/success_at_end': array(0.21875, dtype=float32)}`

3. For RLinf-OpenVLAOFT-GRPO-LIBERO-long:
`eval_metrics={'eval/env_info/success_once': array(0., dtype=float32), 'eval/env_info/return': array(0., dtype=float32), 'eval/env_info/episode_len': array(512., dtype=float32), 'eval/env_info/reward': array(0., dtype=float32), 'eval/env_info/success_at_end': array(0., dtype=float32)}`

These results are significantly lower than the expected performance.

The only modifications I made were:

- Updated the model paths in `examples/embodiment/config/libero_10_grpo_openvlaoft_eval.yaml`
- Changed num_envs: 256 in `examples/embodiment/config/env/train/libero_10_grpo.yaml`

Could you please help investigate why the evaluation results cannot be reproduced as expected? Are there any additional configurations or steps I might be missing?

Thank you for your assistance.

### Environment

- Python version:
Python 3.11.10
- PIP list (grep -E "torch|sglang|vllm|ray|transformers"):
arm_pytorch_utilities              0.4.3
array_record                       0.8.1
pytorch-kinematics                 0.7.5
pytorch-seed                       0.2.0
ray                                2.47.0
torch                              2.5.1+cu124
torch_memory_saver                 0.0.8
torchaudio                         2.5.1+cu124
torchdata                          0.11.0
torchelastic                       0.2.2
torchvision                        0.20.1+cu124
transformers                       4.40.1
- RLinf version: 0.1.0
- Megatron version: not used
- Docker image version: agentic-openvlaoft-rlinf0.1-torch2.5.1
- nvidia-smi:
```
Mon Sep  8 08:10:30 2025       
+---------------------------------------------------------------------------------------+
| NVIDIA-SMI 535.171.04             Driver Version: 535.171.04   CUDA Version: 12.4     |
|-----------------------------------------+----------------------+----------------------+
| GPU  Name                 Persistence-M | Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp   Perf          Pwr:Usage/Cap |         Memory-Usage | GPU-Util  Compute M. |
|                                         |                      |               MIG M. |
|=========================================+======================+======================|
|   0  NVIDIA A40                     Off | 00000000:14:00.0 Off |                  Off |
|  0%   23C    P8              30W / 300W |     13MiB / 49140MiB |      0%      Default |
|                                         |                      |                  N/A |
+-----------------------------------------+----------------------+----------------------+
|   1  NVIDIA A40                     Off | 00000000:15:00.0 Off |                  Off |
|  0%   22C    P8              23W / 300W |     13MiB / 49140MiB |      0%      Default |
|                                         |                      |                  N/A |
+-----------------------------------------+----------------------+----------------------+
|   2  NVIDIA A40                     Off | 00000000:18:00.0 Off |                  Off |
|  0%   22C    P8              27W / 300W |     13MiB / 49140MiB |      0%      Default |
|                                         |                      |                  N/A |
+-----------------------------------------+----------------------+----------------------+
|   3  NVIDIA A40                     Off | 00000000:1D:00.0 Off |                  Off |
|  0%   23C    P8              23W / 300W |     13MiB / 49140MiB |      0%      Default |
|                                         |                      |                  N/A |
+-----------------------------------------+----------------------+----------------------+
|   4  NVIDIA A40                     Off | 00000000:1E:00.0 Off |                  Off |
|  0%   22C    P8              23W / 300W |     13MiB / 49140MiB |      0%      Default |
|                                         |                      |                  N/A |
+-----------------------------------------+----------------------+----------------------+
|   5  NVIDIA A40                     Off | 00000000:21:00.0 Off |                  Off |
|  0%   22C    P8              26W / 300W |     13MiB / 49140MiB |      0%      Default |
|                                         |                      |                  N/A |
+-----------------------------------------+----------------------+----------------------+
|   6  NVIDIA A40                     Off | 00000000:25:00.0 Off |                  Off |
|  0%   22C    P8              23W / 300W |     13MiB / 49140MiB |      0%      Default |
|                                         |                      |                  N/A |
+-----------------------------------------+----------------------+----------------------+
|   7  NVIDIA A40                     Off | 00000000:2D:00.0 Off |                  Off |
|  0%   20C    P8              23W / 300W |     13MiB / 49140MiB |      0%      Default |
|                                         |                      |                  N/A |
+-----------------------------------------+----------------------+----------------------+
                                                                                         
+---------------------------------------------------------------------------------------+
| Processes:                                                                            |
|  GPU   GI   CI        PID   Type   Process name                            GPU Memory |
|        ID   ID                                                             Usage      |
|=======================================================================================|
+---------------------------------------------------------------------------------------+

```

### Reproduction

Already described in Problem description. 

### Before submitting a new issue...

- [x] Have you checked relevant issues, FAQs (https://rlinf-docs.readthedocs.io/en/latest/rst_source/faq.html), or asked the chatbot at the top right corner of the [documentation page](https://rlinf-docs.readthedocs.io).

**Comment by zoeyuchao** [comment]
@WinstonWmj Could U help check this issue?

**Comment by WinstonWmj** [comment]
@Weott Thank you for providing the question and detailed relevant information. After our careful inspection, we found that we have included an incorrect version of `transformers` in this Docker image (agentic-openvlaoft-rlinf0.1-torch2.5.1), which is not compatible with the [openvla-oft model](https://github.com/moojink/openvla-oft/blob/main/pyproject.toml#L50). Therefore, you need to additionally clone [this repository transformers==4.40.1](https://github.com/moojink/transformers-openvla-oft) and reinstall it using the command `pip install /pathto/transformers-openvla-oft`. 

Furthermore, I noticed that you modified the YAML file under the "env/train" subfolder, which is incorrect. If you want to modify the `num_envs` parameter, you need to make changes to the corresponding YAML file in the "env/eval" subfolder for the modification to take effect.

**Comment by Weott** [comment]
@WinstonWmj Thank you for your quick and helpful response! The issue has been successfully resolved.

**Comment by ShimBoi** [comment]
Hi I am also having this issue with recreating the results on Libero_10 using openvla-oft. I am using the `libero_10_ppo_openvlaoft.yaml` configuration but with 128 num_envs in the env/eval/liber_10.yaml. 

The models tested are:

Openvla-oft-SFT-libero10-traj1
Openvla-oft-SFT-libero10-trajall
Haozhan72/openvla-oft-libero10-traj1-rl

The evaluation results I obtained are:

For Openvla-oft-SFT-libero10-traj1:
eval_metrics={'eval/env_info/success_once': array(0., dtype=float32), 'eval/env_info/return': array(0., dtype=float32), 'eval/env_info/episode_len': array(512., dtype=float32), 'eval/env_info/reward': array(0., dtype=float32), 'eval/env_info/success_at_end': array(0., dtype=float32)}

For Openvla-oft-SFT-libero10-trajall:
eval_metrics={'eval/env_info/success_once': array(0.28125, dtype=float32), 'eval/env_info/return': array(1.09375, dtype=float32), 'eval/env_info/episode_len': array(512., dtype=float32), 'eva
l/env_info/reward': array(0.00213623, dtype=float32), 'eval/env_info/success_at_end': array(0.21875, dtype=float32)}.

For Haozhan72/openvla-oft-libero10-traj1-rl:
eval_metrics={'eval/env_info/success_once': array(0., dtype=float32), 'eval/env_info/return': array(0., dtype=float32), 'eval/env_info/episode_len': array(512., dtype=float32), 'eval/env_info/reward': array(0., dtype=float32), 'eval/env_info/success_at_end': array(0., dtype=float32)}

I've followed @WinstonWmj comment on cloning the trnasformers-openvla-oft and installing it, but I am still experiencing the same success rates, although I am not using docker but locally cloned:

```
arm_pytorch_utilities              0.4.3
array_record                       0.8.1
pytorch-kinematics                 0.7.5
pytorch-seed                       0.2.0
ray                                2.49.1
torch                              2.6.0
torch_memory_saver                 0.0.8
torchaudio                         2.2.0
torchdata                          0.11.0
torchvision                        0.21.0
transformers                       4.40.1      /scratch/cluster/jshim12/RLinf/transformers-openvla-oft
```
